lib/sidekiq_unique_jobs/client/middleware.rb

```diff
@@ -63,7 +63,7 @@ def old_unique_for?
 
       def new_unique_for?
         connection do |conn|
-          return conn.set(payload_hash, item['jid'], nx: true, ex: expires_at)
+          return conn.set(payload_hash, item['jid'], nx: true, ex: expires_at) || conn.get(payload_hash) == item['jid']
```

Comment:
I don't think this works - because then a perform_async job cannot run if a perform_in job is scheduled (even if the scheduled job is hours in advance). I'd rather not have a lock on the scheduled queue at all then a behaviour like this.